core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -624,6 +683,42 @@ class CityInfo {
         civInfo.updateDetailedCivResources() // this building could be a resource-requiring one
     }
 
+    private fun demandNewResource() {
+        val candidates = getRuleset().tileResources.values.filter {
+            // 1. Luxury
+            it.resourceType == ResourceType.Luxury &&
+            // 2. Not a city-state only resource eg jewelry
+            !it.hasUnique(UniqueType.CityStateOnlyResource) &&
+            // 3. Not the same as last
+            it.name != demandedResource &&
+            // 4. Not found nearby
+            getCenterTile().getTilesInDistance(3).none { nearTile -> nearTile.resource == it.name }
+        }.shuffled()
+
+        // Keep trying until we get one that actually exists on the map
+        val chosenResource = candidates.firstOrNull { tileMap.values.any { tile -> tile.resource == it.name } }
```

Comment:
Wildly inefficient
I think we save all natural wonders in a lazy transient in the tilemap, we can do the same with resources